src/SpotifyLogin.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react"
+import { generateAuthUrl } from "./utils/spotify"
+import axios from "axios";
+
+export const SpotifyLogin = () => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [username, setUsername] = useState("");
+    const [error, setError] = useState(false);
+
+    useEffect(() => {
+        const token = sessionStorage.getItem("token");
+
+        if (!token)
+            return;
+
+        setIsLoggedIn(true);
+
+        axios
+            .get("https://api.spotify.com/v1/me", {
+                headers: {
+                    "Authorization": `Bearer ${token}`
+                }
+            })
+            .then(res => setUsername(res.data["display_name"]))
+            .catch(err => setError(true));
+    }, [isLoggedIn])
+
+    const initiateSpotifyLogin = () => {
+        const params = {
+            client_id: "962be0d99c5f4098985705f22c488288",
+            redirect_uri: `${window.location.origin}/auth`,
```

Comment:
It's incorrect to use window.location.origin cause it will work only when your app hosted in the domain root -> better to replace with window.location.href to use actual url you are on